components/scream/src/physics/p3/p3_functions_upwind_impl.hpp

```diff
@@ -81,14 +83,31 @@ ::calc_first_order_upwind_step (
     });
 }
 
+template <typename S, typename D>
+template <Int kdir, int nfield>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::generalized_sedimentation (
+  const uview_1d<const Spack>& rho,
+  const uview_1d<const Spack>& inv_rho,
+  const uview_1d<const Spack>& inv_dzq,
+  const MemberType& team,
+  const Int& nk, const Int& k_qxtop, Int& k_qxbot, const Int& kbot, const Scalar& Co_max, Scalar& dt_left, Scalar& prt_accum,
+  const view_1d_ptr_array<Spack, nfield>& fluxes,
+  const view_1d_ptr_array<Spack, nfield>& Vs, // (behaviorally const)
+  const view_1d_ptr_array<Spack, nfield>& rs)
+{
+
```

Comment:
Only a stub for now, calc_upwind was plenty challenging on its own for a single PR.

Reply:
You can leave it here for now, but I think it will be a large enough routine to deserve its own file once you impl it.